common/src/test/java/com/microsoft/identity/common/internal/providers/microsoft/nativeauth/utils/MockApiUtils.kt

```diff
@@ -0,0 +1,85 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.common.internal.providers.microsoft.nativeauth.utils
+
+import com.microsoft.identity.common.java.logging.DiagnosticContext
+import com.microsoft.identity.common.java.logging.IRequestContext
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import org.powermock.reflect.Whitebox
+
+class MockApiUtils {
+    companion object {
+        init {
+            MockApi.create()
+        }
+
+        fun setCorrelationIdHeader(correlationId: String) {
+            val mockDiagnosticContext = mock<DiagnosticContext>()
+            Whitebox.setInternalState(
+                DiagnosticContext::class.java,
+                "INSTANCE",
+                mockDiagnosticContext
+            )
+
+            val mockRequestContext = mock<IRequestContext>()
+            whenever(mockRequestContext[DiagnosticContext.CORRELATION_ID]).thenReturn(correlationId)
+            whenever(mockDiagnosticContext.requestContext).thenReturn(mockRequestContext)
+        }
```

Comment:
Seem to be a lot of use of Mockito. Could've easily used Robolectric and simplified a bunch of logic. Please seriously consider moving your tests to Robolectric as that is our preferred framework for writing unit and integration tests.

Reply:
"as that is our preferred framework for writing unit and integration tests" - why?

Reply:
I mean Robolectric is just better. You can just go read their docs. Also you will see the most existing tests are using Robolectric. You may also read this doc I wrote a while back: https://github.com/AzureAD/android-complete/blob/master/docs/Automation/testing_overview.md#android-auth-library-testing-philosophy